modules/core/src/convert.dispatch.cpp

```diff
@@ -278,8 +313,14 @@ void Mat::convertTo(OutputArray dst, int type_, double alpha, double beta) const
     int dtype = CV_MAKETYPE(ddepth, cn);
 
     Mat src = *this;
-    dst.create(dims, size, dtype);
```

Comment:
The root problem somewhere in `.create` method or above (need some time to find where exactly).

Separately, the following operation reuses memory:
```cpp
Mat m(10, 10, CV_32F);
std::cout << (void*)m.data << std::endl;

OutputArray dst(m);
dst.create(10, 10, CV_32S);
Mat dstMat = dst.getMat();
std::cout << (void*)dstMat.data << std::endl;
```
```
0x5c8479f742c0
0x5c8479f742c0
```